src/Eshopworld.Messaging/Messenger.cs

```diff
@@ -97,14 +110,26 @@ public void Receive<T>(Action<T> callback, int batchSize = 10)
         public async Task Subscribe<T>(Action<T> callback, string subscriptionName, int batchSize = 10)
             where T : class
         {
-            if (!MessageSubs.TryAdd(typeof(T), MessagesIn.OfType<T>().Subscribe(callback)))
+            if (!MessageSubs.TryAdd(GetTypeName<T>(), MessagesIn.OfType<T>().Subscribe(callback)))
             {
                 throw new InvalidOperationException("You already added a callback to this message type. Only one callback per type is supported.");
             }
 
             await ((TopicAdapter<T>)SetupMessageType<T>(batchSize, MessagingTransport.Topic)).StartReading(subscriptionName).ConfigureAwait(false);
         }
 
+        /// <inheritdoc />
+        public async Task Subscribe<T>(Action<T> callback, string subscriptionName, string topicName, int batchSize = 10) where T : class
+        {
+            CheckTopicName(topicName);
+            if (!MessageSubs.TryAdd(topicName, MessagesIn.OfType<T>().Subscribe(callback)))

```

Comment:
I know this is just an extension of existing code more or less but I see one problem:

we allow someone to provide a subscription name while subscribing but we allow them to subscribe only once. If our instance of message-subscription to topics is de facto a subscriber, I would expect that information to be defined on the interface level, not while calling the 'subscribe' method. 

So either we recognise the whole class represents a single subscriber or we allow to have many subscribers to the same topic on it.

Reply:
I would keep the the subscriptionName as a parameter. you could have many subscribers for the same topic, but what this could also enable ( which can be useful) is a competing consumer pattern if multiple consuming application instances share a  subscriptionName allowing them to round robin messages from the same subscription, while other applications are free to set their own subscription pattern for the topic ( essentially allowing 1:1 or 1: * for the consumer.

Reply:
are you proposing to change the current behaviour?  Currently only one subscriber is allowed per topic, even when using the message type for the topic name

Reply:
No changes, no. It is a behaviour you already get implicitly by keeping the subscriptionName in the signature.

Reply:
The only problem is, @C-M-Ryan , that you can't have more than 1 subscriber with this implementation anyway - it simply is not taking subscription name into account, only topic name. So once you register one subscriber, you can't do it with more.

So what I suggested was to either:
- make it explicit (you give the subscription name on the `messanger` level)
or
- use subscription name, along with topic, as your key in the subscriptions dictionary.

As this is an old behaviour, we don't have to change it now but something to consider anyway.